### PR TITLE
refactor GridOrder to Classification, added TaxonGrid and versioning concept

### PR DESCRIPTION
**Pull request by sebbean**
stilli making sure all the old specs work, let me know if there's any obvious gotchyas

### CODE REVIEW

File: app/controllers/spree/admin/grid_orders_controller.rb [hunk 1]
```diff
@@ -4,21 +4,44 @@ class GridOrdersController < BaseController
       include Spree::Admin::GridOrdersHelper
       helper 'spree/products'
       def index
-        @taxons = Spree::Taxon.all
-        if params[:taxon_id]
-          @taxon    = Spree::Taxon.find params[:taxon_id]
-          # searcher  = build_searcher taxon: @taxon.id
-          @products = @taxon.ordered_products
+        @taxons = Spree::Taxon.order(:parent_id)
```
Comment (weeksie) [inline app/controllers/spree/admin/grid_orders_controller.rb:8]: Should probably order by position, non?

File: app/controllers/spree/admin/grid_orders_controller.rb [hunk 2]
```diff
@@ -4,21 +4,44 @@ class GridOrdersController < BaseController
...
         reorder_params = JSON.parse params[:reorder]
-        GridOrder.reorder params[:taxon_id], reorder_params
+        Classification.reorder params[:grid_id], reorder_params
         render json: true
       end
 
+      def create
+        @taxon    = Spree::Taxon.find params[:taxon_id]
+        @taxon.grids.create grid_params
+        binding.pry
```
Comment (weeksie) [inline app/controllers/spree/admin/grid_orders_controller.rb]: OH NOES!

File: app/controllers/spree/admin/grid_orders_controller.rb [hunk 3]
```diff
@@ -4,21 +4,44 @@ class GridOrdersController < BaseController
...
+      end
+
+      def show
+        @grid = TaxonGrid.find params[:id] rescue @taxon.grid
+      end
```
Comment (weeksie) [inline app/controllers/spree/admin/grid_orders_controller.rb:34]: Where is @taxon set? I don't see a before filter and it looks like it'll always be nil here.

File: app/helpers/spree/admin/grid_orders_helper.rb [hunk 4]
```diff
@@ -3,4 +3,8 @@ def taxon_dropdown_menu(taxons, taxon)
     options = options_from_collection_for_select taxons, :id, :pretty_name, taxon.try(:id)
     select_tag :taxon_menu, options, "data-url" => admin_grid_orders_url
   end
+  def grids_dropdown_menu(grids, grid, opt={})
+    options = options_from_collection_for_select grids, :id, :available_on, grid.try(:id)
+    select_tag :grid_id, options, opt #, "data-url" => admin_grid_orders_url
```
Comment (weeksie) [inline app/helpers/spree/admin/grid_orders_helper.rb]: Still using `data-url`? Should we delete it?

File: spec/models/spree/product_decorator_spec.rb [hunk 5]
```diff
@@ -1,28 +1,33 @@
...
+  #   binding.pry
+
+  #   ordered  = [ pos_1, pos_3, pos_4, pos_2 ].map(&:id).zip [ 1, 2, 3, 4 ]
+    
+  #   binding.pry
+
+  #   Spree::Classification.reorder tax_hats.id, ordered
+  #   Spree::Product.in_taxon(tax_hats).map(&:id).should eql [ pos_1, pos_3, pos_4, pos_2 ].map(&:id)
+  # end
 end
```
Comment (weeksie) [inline spec/models/spree/product_decorator_spec.rb]: No more spex?

**Comment by sebbean** [comment]
ok baby i think it's good

**Comment by sebbean** [comment]
do me

**Pull request merged**